File: 프로그래머스/unrated/181930. 주사위 게임 2/주사위 게임 2.cpp
#include <string>
#include <vector>
#include <math.h>
using namespace std;

int solution(int a, int b, int c) {
    if (a != b && a != c && b != c)
        return a + b + c;
    else if (a == b && a == c && b == c)
        return (a + b + c) * (pow(a,2) + pow(b,2) + pow(c,2)) * (pow(a,3) + pow(b,3) + pow(c,3));
    else
        return (a + b + c) * (pow(a,2) + pow(b,2) + pow(c,2));    
}

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.java
class Solution {
    public int solution(int a, int b, int c) {
        if (a != b && b != c && a != c) {
            return a + b + c;
        } else if ((a == b && a != c) || (a == c && a != b) || (b == c && a != b)) {
            return (a + b + c) * (a * a + b * b + c * c);
        } else if (a == b && b == c) {
            return (a + b + c) * (a * a + b * b + c * c) * (a * a * a + b * b * b + c * c * c);
        }
        return 0; 
    }
}
